loader/loader_test.go

```diff
@@ -0,0 +1,423 @@
+		It("returns the requested resource if it exists", func() {
+			returnedApplication := &applicationapiv1alpha1.Application{}
+			err := getObject(application.Name, application.Namespace, k8sClient, ctx, returnedApplication)
+			Expect(err).NotTo(HaveOccurred())
+			Expect(application.Spec).To(Equal(returnedApplication.Spec))
+		})
+
+		It("returns and error if the requested resource doesn't exist", func() {
+			returnedObject := &applicationapiv1alpha1.Application{}
+			err := getObject("non-existent-app", "non-existent-app", k8sClient, ctx, returnedObject)
+			Expect(err).To(HaveOccurred())
+			Expect(errors.IsNotFound(err)).To(BeTrue())
+			Expect(returnedObject).To(Equal(&applicationapiv1alpha1.Application{}))
+		})
+	})
+
+	Context("When calling GetActiveReleasePlanAdmission", func() {
+		It("returns an active release plan admission", func() {
+			returnedObject, err := loader.GetActiveReleasePlanAdmission(ctx, k8sClient, releasePlan)
+			Expect(err).NotTo(HaveOccurred())
+			Expect(returnedObject).NotTo(Equal(&v1alpha1.ReleasePlanAdmission{}))
+			Expect(returnedObject.Name).To(Equal(releasePlanAdmission.Name))
+		})
+
+		It("fails to return an active release plan admission if the target does not match", func() {
+			modifiedReleasePlan := releasePlan.DeepCopy()
+			modifiedReleasePlan.Spec.Target = "non-existent-target"
+
+			returnedObject, err := loader.GetActiveReleasePlanAdmission(ctx, k8sClient, modifiedReleasePlan)
+			Expect(err).To(HaveOccurred())
+			Expect(err.Error()).To(ContainSubstring("no ReleasePlanAdmission found in the target"))
+			Expect(returnedObject).To(BeNil())
+		})
+
+		It("fails to return an active release plan admission if the auto release label is set to false", func() {
+			disabledReleasePlanAdmission := releasePlanAdmission.DeepCopy()
+			disabledReleasePlanAdmission.Labels[v1alpha1.AutoReleaseLabel] = "false"
+			disabledReleasePlanAdmission.Name = "disabled-release-plan-admission"
+			disabledReleasePlanAdmission.ResourceVersion = ""
+			Expect(k8sClient.Create(ctx, disabledReleasePlanAdmission)).To(Succeed())
+
+			Eventually(func() bool {
+				returnedObject, err := loader.GetActiveReleasePlanAdmission(ctx, k8sClient, releasePlan)
+				return returnedObject == nil && err != nil && strings.Contains(err.Error(), "with auto-release label set to false")
+			})
+
+			Expect(k8sClient.Delete(ctx, disabledReleasePlanAdmission)).To(Succeed())
+		})
+	})
+
+	Context("When calling GetActiveReleasePlanAdmissionFromRelease", func() {
+		It("returns an active release plan admission", func() {
+			returnedObject, err := loader.GetActiveReleasePlanAdmissionFromRelease(ctx, k8sClient, release)
+			Expect(err).NotTo(HaveOccurred())
+			Expect(returnedObject).NotTo(Equal(&v1alpha1.ReleasePlanAdmission{}))
+			Expect(returnedObject.Name).To(Equal(releasePlanAdmission.Name))
+		})
+
+		It("fails to return an active release plan admission if the release plan does not match", func() {
+			modifiedRelease := release.DeepCopy()
```

Comment:
this should be deleted like line 95, right?

Reply:
Nope. No object is created here.